src/plugins/intel_cpu/src/transformations/snippets/tpp/op/scalar.hpp

```diff
@@ -0,0 +1,30 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "modifiers.hpp"
+#include "eltwise.hpp"
+#include "snippets/op/reduce.hpp"
+
+namespace ov {
+namespace intel_cpu {
+namespace tpp {
+namespace op {
+
+class Scalar : public ov::snippets::op::Scalar {
```

Comment:
Because Scalar is not a MemoryAccess operation, while all the "conventional" TPP's are.
The `TensorProcessingPrimitive` is derived from `snippets::modifier::MemoryAccess`, and `Scalar` is the only operation in the opset that doesn't need it (because it's not really a TPP - it is processed using custom Xbyak emitter)

Anyway, it's a valid point, I'll add a comment in the code